src/screens/nearest.tsx

```diff
@@ -0,0 +1,288 @@
+import { LocationObject } from 'expo-location'
+import { useState, useEffect } from 'react'
+import { FlatList, StyleSheet, View, Text, Platform } from 'react-native'
+import { Button } from 'react-native-paper'
+import DropDown from 'react-native-paper-dropdown'
+import * as Location from 'expo-location'
+import { config } from '../config'
+
+const tfgmEndpoint = 'https://api.tfgm.com/odata/Metrolinks'
+
+type TfGMData = {
+  AtcoCode: string
+  StationLocation: string
+  Dest0: string
+  Wait0: string
+  Status0: string
+  Carriages0: string
+  Dest1: string
+  Wait1: string
+  Status1: string
+  Carriages1: string
+  Dest2: string
+  Wait2: string
+  Status2: string
+  Carriages2: string
+}[]
+
+type DfTData = {
+  atcoCode: string
+  longitude: string
+  latitude: string
+}
+
+type IncomingTram = {
+  dest: string
+  wait: string
+  carriages: string
+  status: string
+}
+
+class StopData {
+  stopName = ''
+  incomingTrams: IncomingTram[] = []
+}
+
+export default function NearestStopScreen() {
+  const [closestStop, setClosestStop] = useState({ name: '', atcoCode: '' })
+  const [incomingTrams, setIncomingTrams] = useState([] as IncomingTram[])
+  const [stopsObtained, setStopsObtained] = useState(
+    [] as { label: string; value: string }[]
+  )
+
+  async function getAllStops() {
+    if (stopsObtained.length > 0) {
+      return
+    }
+
+    const stops: { label: string; value: string }[] = []
+    const res = await fetch(tfgmEndpoint, {
+      method: 'GET',
+      headers: {
+        'Ocp-Apim-Subscription-Key': config.apiKey,
+      },
+    })
+    const json = (await res.json()) as { value: TfGMData }
+    for (const item of json.value) {
+      const truncAtcoCode = item.AtcoCode.substring(0, item.AtcoCode.length - 1)
+      const newStop = { label: item.StationLocation, value: truncAtcoCode }
+
+      if (
+        !stops.some((value) => {
+          return value.value === truncAtcoCode
+        })
+      ) {
+        stops.push(newStop)
+      }
+    }
+
+    setStopsObtained(stops)
+  }
+
+  async function getNearestStop() {
+    if (closestStop.name !== '') {
+      return
+    }
+
+    const res = await fetch(
+      'https://beta-naptan.dft.gov.uk/Download/MultipleLa',
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: new URLSearchParams({
+          selectedLasNames: 'Greater Manchester / North West (180)',
+          fileTypeSelect: 'csv',
+        }).toString(),
+      }
+    )
+    const stopDataAsCsvDfT = await res.text()
+    let stopDataDfT = csvToJson(stopDataAsCsvDfT)
+    stopDataDfT = filterDuplicates(stopDataDfT)
+  }
+
+  const [location, setLocation] = useState(null as unknown as LocationObject)
+  const [errorMsg, setErrorMsg] = useState('null')
```

Comment:
I'm very uncomfortable how `location` is initialised. Leaving the brackets empty would initiate the state as `undefined` and that's completely fine.

Setting  `errorMsg` to a string with value "null" is very hacky and I wouldn't recommend it.